pgxn/neon/walproposer.c

```diff
@@ -80,20 +80,32 @@ static int	CompareLsn(const void *a, const void *b);
 static char *FormatSafekeeperState(Safekeeper *sk);
 static void AssertEventsOkForState(uint32 events, Safekeeper *sk);
 static char *FormatEvents(WalProposer *wp, uint32 events);
+static void UpdateDonorShmem(WalProposer *wp);
 
 
+#ifdef WALPROPOSER_LIB
+WalproposerShmemState *GetWalpropShmemState()
```

Comment:
I'd like to avoid this. I want to run multiple walproposers in multiple threads (but single process) in simulator eventually, it won't work well with this.

Reply:
Made it thread-local